src/reaktoro_pse/core/util_classes/hessian_functions.py

```diff
@@ -0,0 +1,560 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/reaktoro-pse/"
+#################################################################################
+import numpy as np
+
+from scipy.sparse import coo_matrix
+
+__author__ = "Ilayda Akkor, Alexander V. Dudchenko, Paul Vecchiarelli, Ben Knueven"
+
+
+class HessTypes:
```

Comment:
Would probably make sense for this to be two classes, one for the initial hessian options and the other for the hessian update options.

Reply:
Done, made BFGSInitializationTypes class.